snapcraft/providers.py

```diff
@@ -276,72 +275,65 @@ def prepare_instance(
             target=get_managed_environment_home_path() / ".ssh",
         )
 
-    # set up .bashrc
-    # do not push .bashrc to Multipass instances due to craft-providers limitation
-    # (see https://github.com/canonical/craft-providers/issues/169)
-    if isinstance(instance, MultipassInstance):
-        return
-
-    with tempfile.NamedTemporaryFile(mode="w+t") as bashrc:
-        bashrc.write(
-            # pylint: disable=line-too-long
-            dedent(
-                """\
-                #!/bin/bash
-
-                env_file="$HOME/environment.sh"
-
-                # save default environment on first login
-                if [[ ! -e $env_file ]]; then
-                    env > "$env_file"
-                    sed -i 's/^/export /' "$env_file"      # prefix 'export' to variables
-                    sed -i 's/=/="/; s/$/"/' "$env_file"   # surround values with quotes
-                    sed -i '1i#! /bin/bash\\n' "$env_file" # add shebang
+    # pylint: disable=line-too-long
+    bashrc = dedent(
+        """\
```

Comment:
What about taking the opportunity and moving this content to a module-level constant? everything would be more readable, I guess...